server/observable.go

```diff
@@ -694,7 +694,10 @@ func (o *Observable) sampleAck(sseq, dseq, dcount uint64) {
 		return
 	}
 
-	o.mset.sendq <- &jsPubMsg{o.ackSampleT, o.ackSampleT, _EMPTY_, j, o, 0}
+	// can be nil during server Shutdown but with some ACKs in flight internally
```

Comment:
Do we need a lock here? We should comment either way. Is Observable lock held?

Reply:
Its called by ackMsg which holds the lock already

https://github.com/nats-io/nats-server/blob/4e5638474c94859126676ccc1aa96ebf56bdb760/server/observable.go#L703-L706